torch/_subclasses/meta_utils.py

```diff
@@ -394,12 +398,48 @@ def is_c_of_r(complex_dtype, real_dtype):
 
                 else:
                     is_leaf = safe_is_leaf(t)
-                    sizes, strides, storage_offset = sym_sizes_strides_storage_offset(t)
-                    r = callback(
-                        lambda: torch.empty_strided(
-                            sizes, strides, dtype=t.dtype, device="meta"
-                        )
+                    sizes, strides, storage_offset = sym_sizes_strides_storage_offset(
+                        t, source
                     )
+
+                    # If we have a subclass that desugars into dense tensors,
+                    # perform our callback on each inner tensor.
+                    if is_traceable_wrapper_subclass(t):
+
+                        def empty_create(inner_t, inner_src):
+                            is_leaf = safe_is_leaf(inner_t)
+                            (
+                                inner_sizes,
+                                inner_strides,
+                                inner_storage_offset,
+                            ) = sym_sizes_strides_storage_offset(inner_t, inner_src)
+                            return torch.empty_strided(
+                                inner_sizes,
+                                inner_strides,
+                                dtype=inner_t.dtype,
+                                device="meta",
+                            )
+
+                        # Note: transform_subclass will use __tensor_unflatten__ to generate
+                        # a fresh subclass wrapper, which is why sizes/strides are not passed in
+                        # to the creation function here.
+                        # We assume that if the inner tensors of the subclass are given symbolic sizes,
+                        # their sizes will be used to construct the (symbolic) sizes of the wrapper tensor.
+                        r = transform_subclass(
+                            t,
+                            lambda inner_src, inner_t: callback(
```

Comment:
Is this a botched merge? As best as I can tell you will only get one argument to the callback

Reply:
Darn.. yep, there was also supposed to be an update to `transform_subclass()` in this PR, that passes every (key, value) from the dict returned by __tensor_flatten__ (I accidentally committed it to a higher PR in the stack - I ran some tests for sanity before pushing, but definitely messed something up). I'm going to:

(1) move the transform_subclass() changes into this PR

(2) change it so that tensor_flatten only has to return keys (the attribute names corresponding to tensors).